apollo-federation/src/link/federation_spec_definition.rs

```diff
@@ -469,15 +469,16 @@ impl FederationSpecDefinition {
         })
     }
 
-    pub(crate) fn context_directive_arguments(
```

Comment:
Refactors:
- We generally provide `self` here, as a directive's definition, whether it should be present/absent, or its processing/validations may depend on spec version.

Reply:
Mainly it's just to prevent people from forgetting to account for spec version in these kinds of functions (or forgoing it because its too much extra work).

As an example, imagine in a future spec bump that this directive allows users to specify more information in a backwards compatible way (e.g. a new argument, or a new syntax for an existing string-ish argument). It would be very tempting in such cases to keep the method static, and to just use the "latest" directive's semantics to parse (after all, its "backwards compatible"). However, this leads to bugs where code can accidentally permit that new information with old specs when it shouldn't. Some of this is prevented by the GraphQL type system itself (e.g. new arguments via directive definition validation), but things beyond that type system won't be (e.g. new syntaxes for existing string-ish arguments).